hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/execution/traceability/SolidityAction.java

```diff
@@ -52,6 +57,53 @@ public SolidityAction(
         this.callDepth = callDepth;
     }
 
+    /**
+     * Check that the fields are set correctly for the semantics of the underlying protobuf
+     *
+     * Respecting, for example, protobuf `oneof` unions... (but allowing for `oneof recipient` to be missing
+     * entirely, because that's ok)
+     */
+    public boolean isValid() {
```

Comment:
Would it not be simpler to convert to a gRPC object and check the resulting object via `has*()` methods?